src/agora/agora.cc

```diff
@@ -205,54 +205,36 @@ void Agora::ScheduleAntennasTX(size_t frame_id, size_t symbol_id) {
 
 void Agora::ScheduleSubcarriers(EventType event_type, size_t frame_id,
                                 size_t symbol_id) {
-  auto base_tag = gen_tag_t::FrmSymSc(frame_id, symbol_id, 0);
-  size_t num_events = SIZE_MAX;
-  size_t block_size = SIZE_MAX;
+  gen_tag_t base_tag(0);
+  size_t num_events;
+  size_t block_size;
 
   switch (event_type) {
     case EventType::kDemul:
-    case EventType::kPrecode:
+    case EventType::kPrecode: {
+      base_tag = gen_tag_t::FrmSymSc(frame_id, symbol_id, 0);
       num_events = config_->DemulEventsPerSymbol();
       block_size = config_->DemulBlockSize();
       break;
-    case EventType::kBeam:
-      num_events = config_->BeamCallsPerSymbol() / config_->BeamBatchSize();
+    }
+    case EventType::kBeam: {
+      base_tag = gen_tag_t::FrmSc(frame_id, 0);
+      num_events = config_->BeamEventsPerSymbol();
       block_size = config_->BeamBlockSize();
       break;
-    default:
-      assert(false);
-  }
-
-  size_t qid = (frame_id & 0x1);
-  if (event_type == EventType::kBeam) {
-    EventData event;
-    event.event_type_ = event_type;
-    event.num_tags_ = config_->BeamBatchSize();
```

Comment:
What was the purpose to use a batch of calls in a single event?

Reply:
No clue, but it looks redundant to me.   If we want to make the calls sequential we can just change the equation for BeamBlockSize